File: backend-php/controllers/MessageController.php
<?php
/**
 * Contrôleur messages : /api/messages, /api/messages/conversation.
 * Équivalent PHP de com.ihecride.controllers.MessageController.
 */

require_once __DIR__ . '/../dao/MessageDAO.php';
require_once __DIR__ . '/../models/Message.php';

class MessageController {

    private MessageDAO $dao;

    public function __construct() {
        $this->dao = new MessageDAO();
    }

    public function conversation(array $ctx): void {
        $other = (int) ($_GET['userId'] ?? 0);
        $list  = $this->dao->findConversation($ctx['userId'], $other);
        $this->dao->marquerLu($ctx['userId'], $other);
        json_send(['success' => true, 'data' => $list]);
    }

    public function send(array $ctx): void {
        $m = new Message();
        $m->expediteurId   = $ctx['userId'];
        $m->destinataireId = (int) ($_POST['destinataireId'] ?? 0);
        $m->contenu        = $_POST['contenu'] ?? '';
        $id = $this->dao->insert($m);
        json_send(['success' => $id > 0, 'id' => $id]);
    }
}

File: backend/src/main/java/com/ihecride/controllers/MessageController.java
package com.ihecride.controllers;

import com.ihecride.dao.MessageDAO;
import com.ihecride.models.Message;

import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.List;

@WebServlet({"/api/messages", "/api/messages/conversation"})
public class MessageController extends HttpServlet {

    private final MessageDAO dao = new MessageDAO();

    @Override
    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
        resp.setContentType("text/plain; charset=UTF-8");
        int userId = (int) req.getAttribute("userId");
        int other = Integer.parseInt(req.getParameter("userId"));
        List<Message> list = dao.findConversation(userId, other);
        dao.marquerLu(userId, other);
        try (PrintWriter out = resp.getWriter()) {
            out.println("id|expediteurId|expediteurNom|contenu|dateEnvoi|lu");
            for (Message m : list) {
                out.println(m.getId() + "|" + m.getExpediteurId() + "|" +
                            safe(m.getExpediteurNom()) + "|" +
                            safe(m.getContenu()) + "|" + m.getDateEnvoi() + "|" + m.isLu());
            }
        }
    }

    @Override
    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
        req.setCharacterEncoding("UTF-8");
        int userId = (int) req.getAttribute("userId");
        Message m = new Message();
        m.setExpediteurId(userId);
        m.setDestinataireId(Integer.parseInt(req.getParameter("destinataireId")));
        m.setContenu(req.getParameter("contenu"));
        int id = dao.insert(m);
        resp.getWriter().write("success=" + (id > 0) + "&id=" + id);
    }

    private String safe(String s) { return s == null ? "" : s.replace("|", "/").replace("\n"," "); }
}
